src/widgets/block.rs

```diff
@@ -887,4 +887,15 @@ mod tests {
                 .remove_modifier(Modifier::DIM)
         )
     }
+
+    #[test]
+    fn title_is_aligned_on_center() {
+        assert_eq!(
+            Block::default()
+                .title("test")
+                .title_alignment(Alignment::Center)
+                .titles_alignment,
+            Alignment::Center
+        );
+    }
```

Comment:
This test will still succeed if the `.title()` line is removed.
How about the following which tests the functionality rather than just the property:
```suggestion
#[test]
    fn title_alignment() {
        let mut buffer = Buffer::empty(Rect::new(0, 0, 8, 1));
        let tests = vec![
            (Alignment::Left, "test    "),
            (Alignment::Center, "  test  "),
            (Alignment::Right, "    test"),
        ];

        for (alignment, expected) in tests {
            Block::default()
                .title("test")
                .title_alignment(alignment)
                .render(buffer.area, &mut buffer);
            assert_buffer_eq!(buffer, Buffer::with_lines(vec![expected]));
        }
    }

    #[test]
    fn title_alignment_overrides_block_title_alignment() {
        let mut buffer = Buffer::empty(Rect::new(0, 0, 8, 1));
        let tests = vec![
            // (block.title_alignment, title.alignment, expected)
            (Alignment::Right, Alignment::Left, "test    "),
            (Alignment::Left, Alignment::Center, "  test  "),
            (Alignment::Center, Alignment::Right, "    test"),
        ];

        for (block_title_alignment, alignment, expected) in tests {
            Block::default()
                .title(Title::clone_from("test").alignment(alignment))
                .title_alignment(block_title_alignment)
                .render(buffer.area, &mut buffer);
            assert_buffer_eq!(buffer, Buffer::with_lines(vec![expected]));
        }
    }
```
Incidentally this highlights a bug that the title part doesn't fully erase the background behind the title.:
```text
expected: Buffer {
    area: Rect { x: 0, y: 0, width: 8, height: 1 },
    content: [
        "  test  ",
    ],
    styles: [
        x: 0, y: 0, fg: Reset, bg: Reset, underline: Reset, modifier: NONE,
    ]
}
actual: Buffer {
    area: Rect { x: 0, y: 0, width: 8, height: 1 },
    content: [
        "tetest  ",
    ],
    styles: [
        x: 0, y: 0, fg: Reset, bg: Reset, underline: Reset, modifier: NONE,
    ]
}
```

Reply:
Fair point.